salt/modules/virt.py

```diff
@@ -3213,6 +3377,107 @@ def _set_vcpu(node, value):
+        {"path": "cpu:tuning:quota", "xpath": "cputune/quota"},
+        {"path": "cpu:tuning:global_period", "xpath": "cputune/global_period"},
+        {"path": "cpu:tuning:global_quota", "xpath": "cputune/global_quota"},
+        {"path": "cpu:tuning:emulator_period", "xpath": "cputune/emulator_period"},
+        {"path": "cpu:tuning:emulator_quota", "xpath": "cputune/emulator_quota"},
+        {"path": "cpu:tuning:iothread_period", "xpath": "cputune/iothread_period"},
+        {"path": "cpu:tuning:iothread_quota", "xpath": "cputune/iothread_quota"},
+        {
+            "path": "cpu:tuning:vcpupin:{id}",
+            "xpath": "cputune/vcpupin[@vcpu='$id']",
+            "get": lambda n: str(n.get("cpuset")) if n.get("cpuset") else None,
+            "set": lambda n, v: n.set("cpuset", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("cpuset", ["vcpu"]),
+        },
+        {
+            "path": "cpu:tuning:emulatorpin",
+            "xpath": "cputune/emulatorpin",
+            "get": lambda n: str(n.get("cpuset")) if n.get("cpuset") else None,
+            "set": lambda n, v: n.set("cpuset", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("cpuset", []),
+        },
+        {
+            "path": "cpu:tuning:iothreadpin:{id}",
+            "xpath": "cputune/iothreadpin[@iothread='$id']",
+            "get": lambda n: str(n.get("cpuset")) if n.get("cpuset") else None,
+            "set": lambda n, v: n.set("cpuset", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("cpuset", ["iothread"]),
+        },
+        {
+            "path": "cpu:tuning:vcpusched:{id}:scheduler",
+            "xpath": "cputune/vcpusched[@vcpus='$id']",
+            "get": lambda n: str(n.get("scheduler")) if n.get("scheduler") else None,
+            "set": lambda n, v: n.set("scheduler", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("scheduler", ["vcpus"]),
+        },
+        {
+            "path": "cpu:tuning:vcpusched:{id}:priority",
+            "xpath": "cputune/vcpusched[@vcpus='$id']",
+            "get": lambda n: str(n.get("priority")) if n.get("priority") else None,
+            "set": lambda n, v: n.set("priority", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("priority", ["vcpus"]),
+        },
+        {
+            "path": "cpu:tuning:iothreadsched:iothreads",
+            "xpath": "cputune/iothreadsched",
+            "get": lambda n: str(n.get("iothreads")) if n.get("iothreads") else None,
+            "set": lambda n, v: n.set("iothreads", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("iothreads", []),
+        },
+        {
+            "path": "cpu:tuning:iothreadsched:scheduler",
+            "xpath": "cputune/iothreadsched",
+            "get": lambda n: str(n.get("iothreads")) if n.get("scheduler") else None,
+            "set": lambda n, v: n.set("scheduler", str(v)),
+            "del": salt.utils.xmlutil.del_attribute("scheduler", []),
+        },
```

Comment:
As per the [schema](https://gitlab.com/libvirt/libvirt/-/blob/master/docs/schemas/domaincommon.rng#L966) there can be mulitple `iothreadsched`... they need to be a list

You can also add the `emulatorsched` (not a list)